pkg/apis/sources/v1alpha1/jobsource_validation.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// Validate implements apis.Validatable
+func (js *JobSource) Validate(ctx context.Context) *apis.FieldError {
```

Comment:
our customm webhook calls this method fyi. If the webhook is not installed and running, this will never do anything.